src/Database.m

```diff
@@ -134,7 +134,7 @@ - (BOOL)initialiseDatabase {
     } else if (databaseVersion >= MA_Min_Supported_DB_Version) {
         NSAlert * alert = [[NSAlert alloc] init];
         [alert setMessageText:NSLocalizedString(@"Database Upgrade", nil)];
-        [alert setInformativeText:NSLocalizedString(@"Vienna must upgrade its database to the latest version. This may take a minute or so. We apologize for the inconveninece.", nil)];
+        [alert setInformativeText:NSLocalizedString(@"Database upgrade explanation text", @"Vienna must upgrade its database to the latest version. This may take a minute or so. We apologize for the inconvenience.")];
```

Comment:
I do not agree with the approach suggested in that article. NSLocalizedString already supports contextualisation by using the "comment" and "table" parameters (which Vienna hardly ever uses either). The author actually reinforces the bad behaviour of not specifying a meaningful comment in the first place, instead making up for it by using abstract localisation keys, adding even more complexity by making translation keys themselves utterly useless.

XLIFF export/import seems to be a remarkably simple way of managing strings files. No more obsolete keys and keys are properly commented (if they have a comment field, which they should). XLIFF is also a much more portable format, allowing translators to modify them.